public/app/plugins/datasource/loki/querybuilder/parsing.ts

```diff
@@ -409,3 +424,26 @@ function getLastChildWithSelector(node: SyntaxNode, selector: string) {
   }
   return child;
 }
+
+/**
+ * Get all nodes with type in the tree. This traverses the tree so it is safe only when you know there shouldn't be
+ * too much nesting but you just want to skip some of the wrappers. For example getting function args this way would
+ * not be safe is it would also find arguments of nested functions.
+ * @param expr
+ * @param cur
+ * @param type
+ */
+function getAllByType(expr: string, cur: SyntaxNode, type: string): string[] {
```

Comment:
I think we can just move this to `prometheus/shared/parsingUtils` and use it from there.

Reply:
done in https://github.com/grafana/grafana/pull/47257/commits/d52f1290a789b5eddc5f56ddd420f4ecbdfaffaa